moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame.cpp

```diff
@@ -612,13 +612,20 @@ void MotionPlanningFrame::initFromMoveGroupNS()
   planning_scene_world_publisher_ =
       node_->create_publisher<moveit_msgs::msg::PlanningSceneWorld>("planning_scene_world", 1);
 
+  // Declare parameter for default planning pipeline
+  if (!node_->has_parameter(planning_display_->getMoveGroupNS() + "default_planning_pipeline"))
+    node_->declare_parameter<std::string>(planning_display_->getMoveGroupNS() + "default_planning_pipeline", "");
+
   // Set initial velocity and acceleration scaling factors from ROS parameters
   double factor;
   node_->get_parameter_or("robot_description_planning.default_velocity_scaling_factor", factor, 0.1);
   ui_->velocity_scaling_factor->setValue(factor);
   node_->get_parameter_or("robot_description_planning.default_acceleration_scaling_factor", factor, 0.1);
   ui_->acceleration_scaling_factor->setValue(factor);
 
+  // Query default planning pipeline id
+  node_->get_parameter(planning_display_->getMoveGroupNS() + "default_planning_pipeline", default_planning_pipeline_);
```

Comment:
I still think this should get moved up to about L618 so it is together with the `default_planning_pipeline` parameter declaration. Is there a reason not to do that?

Reply:
Ah, I misunderstood your previous comment - I thought you were asking to put it with the other get_parameter lines. Moved the declare and the get to be more close.